react/react1/week3/todo-tracker/src/App.js

```diff
@@ -0,0 +1,122 @@
+//here two useState(toDoList) are set because once the search is set the
+// fulltodo list gets modified and shorten. Then the other functionalities will be
+// effective on the shorten list once the search bar is cleared. So, the second
+//useState(toDoList) is used to get the full todolist once the search bar is cleared.
+import "./App.css";
+import React, { useEffect } from "react";
+import ToDoList from "./components/ToDoList";
+import Search from "./components/Search";
+import AddToDo from "./components/AddToDo";
+import WatchCount from "./components/WatchCount";
+
+function App() {
+  const [filterToDoList, setFilterToDoList] = React.useState([]);
+  const [fullToDoList, setFullToDoList] = React.useState([]);
+  const [arrLength, setArrLength] = React.useState();
+  useEffect(() => {
+    fetch(
+      "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+    )
+      .then((res) => res.json())
+      .then(
+        (data) => {
+          setArrLength(data.length);
+          data.forEach((e) => (e.taskCompletionStatus = "false"));
+          setFilterToDoList(data);
+          setFullToDoList(data);
+        },
+        (error) => {
+          console.log(error);
+        }
+      );
+  }, []);
+
+  return (
+    <div className="App">
+      <WatchCount />
+
+      <Search
+        onFilteredToDoListChange={(text) => {
+          if (text) {
+            setFilterToDoList(
+              filterToDoList.filter((tododata) =>
+                tododata.description.toLowerCase().includes(text.toLowerCase())
+              )
+            );
+          } else {
+            setFilterToDoList(fullToDoList);
+          }
+        }}
+      />
+
+      <AddToDo
+        onAddTodoList={(task, time) => {
+          const newId = parseInt(fullToDoList.length) + arrLength;
+          const newObj = {
+            id: "",
+            description: "",
+            deadline: "",
+            taskCompletionStatus: "",
+          };
+          newObj.id = (newId + 1).toString();
+          newObj.description = task.value;
+          newObj.deadline = time.value;
+          newObj.taskCompletionStatus = "false";
```

Comment:
Try to set this boolean value in the state. If you want to use the boolean false, in this case it will always return false because the boolean type false is not equal to the string "false".